src/compile.rs

```diff
@@ -341,6 +344,14 @@ fn compile_target(
         build_command.env("PYO3_CROSS_LIB_DIR", lib_dir);
     }
 
+    // pass context for cross-compiles with C/C++ libraries
+    for key in C_LIBRARIES_BUILD_ENVIRONMENT {
+        if let Ok(val) = env::var(key) {
```

Comment:
This could also iterate through and look for `CHOST_<target>` and similar vars for cross-compilation I suppose, but I'm not really familiar.

Reply:
Indeed, you can refer to cc-rs: https://github.com/rust-lang/cc-rs/blob/f2e1b1c9ff92ad063957382ec445bc54e9570c71/src/lib.rs#L2354-L2372